pkg/util/containers/v2/metrics/collector_docker_linux.go

```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+//go:build docker && linux
+// +build docker,linux
+
+package metrics
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/docker/docker/api/types"
+)
+
+func convertContainerStats(stats *types.Stats) *ContainerStats {
+	return &ContainerStats{
+		Timestamp: time.Now(),
+		CPU:       convertCPUStats(&stats.CPUStats),
+		Memory:    convertMemoryStats(&stats.MemoryStats),
+		IO:        convertIOStats(&stats.BlkioStats),
+		PID:       convertPIDStats(&stats.PidsStats),
+	}
+}
+
+func convertCPUStats(cpuStats *types.CPUStats) *ContainerCPUStats {
+	return &ContainerCPUStats{
+		Total:            util.Float64Ptr(float64(cpuStats.CPUUsage.TotalUsage)),
+		System:           util.Float64Ptr(float64(cpuStats.CPUUsage.UsageInKernelmode)),
+		User:             util.Float64Ptr(float64(cpuStats.CPUUsage.UsageInUsermode)),
+		ThrottledPeriods: util.Float64Ptr(float64(cpuStats.ThrottlingData.ThrottledPeriods)),
+		ThrottledTime:    util.Float64Ptr(float64(cpuStats.ThrottlingData.ThrottledTime)),
+	}
+}
+
+func convertMemoryStats(memStats *types.MemoryStats) *ContainerMemStats {
+	containerMemStats := &ContainerMemStats{
+		UsageTotal: util.Float64Ptr(float64(memStats.Usage)),
+		Limit:      util.Float64Ptr(float64(memStats.Limit)),
+	}
+
+	if rss, found := memStats.Stats["rss"]; found {
+		containerMemStats.RSS = util.Float64Ptr(float64(rss))
+	}
+
+	if cache, found := memStats.Stats["cache"]; found {
+		containerMemStats.Cache = util.Float64Ptr(float64(cache))
+	}
+
+	return containerMemStats
+}
+
+func convertIOStats(ioStats *types.BlkioStats) *ContainerIOStats {
+	containerIOStats := ContainerIOStats{
+		ReadBytes:       util.Float64Ptr(0),
+		WriteBytes:      util.Float64Ptr(0),
+		ReadOperations:  util.Float64Ptr(0),
+		WriteOperations: util.Float64Ptr(0),
+		Devices:         make(map[string]DeviceIOStats),
+	}
+
+	for _, blkioStatEntry := range ioStats.IoServiceBytesRecursive {
+		deviceName := fmt.Sprintf("%d:%d", blkioStatEntry.Major, blkioStatEntry.Minor)
```

Comment:
We tend to avoid reporting `major:minor` as device name and report the device name using the mapping.
The idea would be to see if we can re-use the `getDiskDeviceMapping` code in `pkg/util/containers/v2/metrics/collector_disk_linux.go`.

Reply:
Makes sense !
`convertIOStats` is now leveraging `getDiskDeviceMapping`.